Mathlib/Analysis/NormedSpace/ProdLp.lean

```diff
@@ -0,0 +1,775 @@
+
+/-- `WithLp.equiv` as a continuous linear equivalence. -/
+@[simps! (config := { fullyApplied := false }) apply symm_apply]
+protected def prodContinuousLinearEquiv : WithLp p (α × β) ≃L[𝕜] α × β where
+  toLinearEquiv := WithLp.linearEquiv _ _ _
+  continuous_toFun := prod_continuous_equiv _ _ _
+  continuous_invFun := prod_continuous_equiv_symm _ _ _
+
+end ContinuousLinearEquiv
+
+/-! Throughout the rest of the file, we assume `1 ≤ p` -/
+variable [hp : Fact (1 ≤ p)]
+
+/-- `PseudoEMetricSpace` instance on the product of two pseudoemetric spaces, using the
+`L^p` pseudoedistance, and having as uniformity the product uniformity. -/
+instance instProdPseudoEMetricSpace [PseudoEMetricSpace α] [PseudoEMetricSpace β] :
+    PseudoEMetricSpace (WithLp p (α × β)) :=
+  (prodPseudoEMetricAux p α β).replaceUniformity (prod_aux_uniformity_eq p α β).symm
+
+/-- `EMetricSpace` instance on the product of two emetric spaces, using the `L^p`
+edistance, and having as uniformity the product uniformity. -/
+instance instProdEMetricSpace [EMetricSpace α] [EMetricSpace β] : EMetricSpace (WithLp p (α × β)) :=
+  EMetricSpace.ofT0PseudoEMetricSpace (WithLp p (α × β))
+
+/-- `PseudoMetricSpace` instance on the product of two pseudometric spaces, using the
+`L^p` distance, and having as uniformity the product uniformity. -/
+instance instProdPseudoMetricSpace [PseudoMetricSpace α] [PseudoMetricSpace β] :
+    PseudoMetricSpace (WithLp p (α × β)) :=
+  ((prodPseudoMetricAux p α β).replaceUniformity
+    (prod_aux_uniformity_eq p α β).symm).replaceBornology
+    fun s => Filter.ext_iff.1 (prod_aux_cobounded_eq p α β).symm sᶜ
+
+/-- `MetricSpace` instance on the product of two metric spaces, using the `L^p` distance,
+and having as uniformity the product uniformity. -/
+instance instProdMetricSpace [MetricSpace α] [MetricSpace β] : MetricSpace (WithLp p (α × β)) :=
+  MetricSpace.ofT0PseudoMetricSpace _
+
+variable {p α β}
+
+theorem prod_nndist_eq_add [PseudoMetricSpace α] [PseudoMetricSpace β]
+    (hp : p ≠ ∞) (x y : WithLp p (α × β)) :
+    nndist x y = (nndist x.fst y.fst ^ p.toReal + nndist x.snd y.snd ^ p.toReal) ^ (1 / p.toReal) :=
+  NNReal.eq <| by
+    push_cast
+    exact prod_dist_eq_add (p.toReal_pos_iff_ne_top.mpr hp) _ _
+
+theorem prod_nndist_eq_sup [PseudoMetricSpace α] [PseudoMetricSpace β] (x y : WithLp ∞ (α × β)) :
+    nndist x y = nndist x.fst y.fst ⊔ nndist x.snd y.snd :=
+  NNReal.eq <| by
+    push_cast
+    exact prod_dist_eq_sup _ _
+
+variable (p α β)
+
+theorem prod_lipschitzWith_equiv [PseudoEMetricSpace α] [PseudoEMetricSpace β] :
+    LipschitzWith 1 (WithLp.equiv p (α × β)) :=
+  prod_lipschitzWith_equiv_aux p α β
+
+theorem prod_antilipschitzWith_equiv [PseudoEMetricSpace α] [PseudoEMetricSpace β] :
+    AntilipschitzWith ((2 : ℝ≥0) ^ (1 / p).toReal) (WithLp.equiv p (α × β)) :=
+  prod_antilipschitzWith_equiv_aux p α β
+
+theorem infty_equiv_isometry [PseudoEMetricSpace α] [PseudoEMetricSpace β] :
```

Comment:
I don't really like this name, as it is not obvious it refers to a prod situation and not a pi situation.